src/txn/txn_ckpt.c

```diff
@@ -830,10 +830,10 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
     __wt_writeunlock(session, &txn_global->rwlock);
 
     /*
-     * Refresh our snapshot here without publishing our shared ids to the world, doing so prevents
-     * us from racing with the stable timestamp moving ahead of current snapshot. i.e. if the stable
-     * timestamp moves after we begin the checkpoint transaction but before we set the checkpoint
-     * timestamp we can end up missing updates in our checkpoint.
+     * Refresh our snapshot here, doing so prevents us from racing with the stable timestamp moving
+     * ahead of current snapshot. i.e. if the stable timestamp moves after we begin the checkpoint
+     * transaction but before we set the checkpoint timestamp we can end up missing updates in our
+     * checkpoint. Call the bump variant as we don't want to publish the relevant ids.
```

Comment:
We want to avoid use publish but here we use it. However, I think it is OK here.

Reply:
No we've decided the "publish" just means a write to shared memory.

Reply:
Then if it is not OK, should we change then?

Reply:
As discussed publish means a write to shared memory.